_posts/2017-02-09-rapid-mysql-imports-docker.md

```diff
@@ -0,0 +1,186 @@
+--
+
+LOCK TABLES `block_content` WRITE;
+/*!40000 ALTER TABLE `block_content` DISABLE KEYS */;
+set autocommit=0;
+INSERT INTO `block_content` VALUES (1,1,'basic','a9167ea6-c6b7-48a1-ac06-6d04a67a5d54','en'),(2,2,'basic','2114eee9-1674-4873-8800-aaf06aaf9773','en'),(3,3,'basic','855c13ba-689e-40fd-9b00-d7e3dd7998ae','en'),(4,4,'basic','8c68671b-715e-457d-a497-2d38c1562f67','en'),(5,5,'basic','bc7701dd-b31c-45a6-9f96-48b0b91c7fa2','en'),(6,6,'basic','d8e23385-5bda-41da-8e1f-ba60fc25c1dc','en'),(7,7,'basic','ea6a93eb-b0c3-4d1c-8690-c16b3c52b3f1','en'),(8,8,'basic','3d314051-567f-4e74-aae4-a8b076603e44','en'),(9,9,'basic','2ef5ae05-6819-4571-8872-4d994ae793ef','en'),(10,10,'basic','3deaa1a9-4144-43cc-9a3d-aeb635dfc2ca','en'),(11,11,'basic','d57e81e8-c613-45be-b1d5-5844ba15413c','en');
+/*!40000 ALTER TABLE `block_content` ENABLE KEYS */;
+UNLOCK TABLES;
+commit;
+```
+
+When I pipe the contents of the MySQL database dump to the `mysql` command, the `mysql` client processes each of these instructions sequentially in order to (1) create the structure for each table defined in the file and (2) populate the database with data from the SQL dump.
+
+What happens when `mysql` finished processing the SQL dump? The database contents live in `/var/lib/mysql/{database}`, so for example for the `block_content` table mentioned above there are two files called `block_content.frm` and `block_content.ibd` in `/var/lib/mysql/{database}/`. The `/var/lib/mysql` directory will also contain a number of other directories and files related to the configuration of the MySQL server.
+
+Now, suppose that, instead of sequentially processing the SQL instructions contained in a database dump file, we were able to provide developers with a snapshot of the `/var/lib/mysql` directory for a given Drupal site. Would this be faster than the traditional database import methods?
+
+The answer is, yes, very much so!
+
+### Two test cases
+
+The table below shows the results of two test cases, one using a 19 MB database file and the other using a 4.7 GB database.
+
+| Method                | Database size | Time to drop tables and restore (seconds) |
+| :-------              | ------:       |                                  -------: |
+| Traditional `mysql`   | 19 MB         |                                       128 |
+| Docker volume restore | 19 MB         |                                        11 |
+| Traditional `mysql`   | 4.7 GB        |                                       606 |
+| Docker volume restore | 4.7 GB        |                                        85 |
+
+In other words, the MySQL data volume import completes, on average, in about 11% of the time that a traditional MySQL dump import would take!
+
+Since a GIF is worth a thousand words, compare these two processes side-by-side (both are using the same source 19 MB source database; the first is using a data volume restore process while the second is using the traditional MySQL import process):
```

Comment:
Consider (my changes are in bold so it's easier for you to see. not saying they should be bold in the post)

"Since a GIF is worth a thousand words, **let's** compare these two processes side-by-side (both are using the same source 19 MB source database; the first is using a data volume restore process while the second is using the traditional MySQL import process). **You can see the 2nd process takes considerably longer!**"